test/canonicalizer_tag_categories_test.rb

```diff
@@ -74,5 +80,44 @@ def test_single_skill_multiple_names
       Canonicalizer.canonicalize_tag_category site.data['skills']
       assert_equal ruby_consolidated, site.data['skills']
     end
+
+    def all_consolidated
+      c_plus_plus.merge(ruby_uppercase).merge(ruby_lowercase)
+    end
+
+    def expected_consolidated
+      ruby_consolidated.merge('c++' => c_plus_plus['C++'])
+    end
+
+    def test_multiple_skills
+      site.data['skills'] = all_consolidated
+      Canonicalizer.canonicalize_tag_category site.data['skills']
+      assert_equal expected_consolidated, site.data['skills']
+    end
+
+    SELF_URL = 'https://team-api.18f.gov/api/skills/'
```

Comment:
Factored out a `CanonicalizeTagTestData` class, as well as a new `CanonicalizeTagsForItemTest` class, and hopefully cleared up the structure all around.